gateway/src/lib.rs

```diff
@@ -44,14 +50,65 @@ pub fn run_openapi() -> Result<(), String> {
 
 pub struct MgsArguments {
     pub id: Uuid,
-    pub address: SocketAddrV6,
+    pub addresses: Vec<SocketAddrV6>,
 }
 
+type HttpServer = dropshot::HttpServer<Arc<ServerContext>>;
+
 pub struct Server {
     /// shared state used by API request handlers
     pub apictx: Arc<ServerContext>,
-    /// dropshot server for requests from nexus
-    pub http_server: dropshot::HttpServer<Arc<ServerContext>>,
+    /// dropshot servers for requests from nexus or wicketd, keyed by their bind
+    /// address
+    pub http_servers: HashMap<SocketAddrV6, HttpServer>,
```

Comment:
There's pretty tight coupling between `http_servers` and `all_servers_shutdown`, right?

Should we be exposing `http_servers` as `pub`? Seems like that risks a caller adding a server without waiting on it, or removing a server without awaiting termination.

Reply:
There is, yes. I don't think any of `Server`'s fields ought to be pub, generally, but when I tried making them private I bumped into issues with the integration tests poking at things, and I put pub back for expedience. I'll see if I can make these private and expose more robust pub methods that the tests can use.

Reply:
Fixed in 876a610a: the fields are now private, and `Server` grew a few methods to support integration tests that don't expose us to someone mucking about with `http_servers`/`all_servers_shutdown` directly.